### ISSUE TITLE
SSLConfigLoad: Failed to load OpenSSL config file.

### ISSUE THREAD

**Issue by binbon2d** [badjtsx/ShrinkLDPlayer#2]
E:\VMWARE\LDPlayer\ShrinkLDPlayer>"E:\VMWARE\LDPlayer\ShrinkLDPlayer\VdiskManager\vmware-vdiskmanager.exe" -k "E:\VMWARE\LDPlayer\LDPlayer9\vms\leidian1\sdcard.vmdk"
SSLConfigLoad: Failed to load OpenSSL config file.
Failed to open the disk 'E:\VMWARE\LDPlayer\LDPlayer9\vms\leidian1\sdcard.vmdk' : The operation was canceled (0xa).
Failed to shrink the disk 'E:\VMWARE\LDPlayer\LDPlayer9\vms\leidian1\sdcard.vmdk' : The operation was canceled (0xa).

E:\VMWARE\LDPlayer\ShrinkLDPlayer>pause
Press any key to continue . . .

i checked all the paths are correct, but it gives the above error
LdPlayer version 9.1.20.1(64)

**Comment by badjtsx** [comment]
Make sure to exit LDPlayer before running the shrink bat file.

**Comment by binbon2d** [comment]
Yes i missing ít but still 
SSLConfigLoad: Failed to load OpenSSL config file
and 100% shink 
But not work
![image](https://github.com/user-attachments/assets/9ba1f6b2-16c3-48d8-8dc3-a1cb5a62fdc5)
I edit : adb shell "su -c 'cd /sdcard && dd if=/dev/zero of=fillfile bs=1048576 count=20480'"

**Comment by badjtsx** [comment]
Can you show me your disk info? 
This page i mean:
![Screenshot (162)](https://github.com/user-attachments/assets/2a265f3f-6d71-4a40-abd8-b92f9a9b2233)

**Comment by binbon2d** [comment]
ok i try again my disk info : Shared read-only

**Comment by badjtsx** [comment]
and the free space? Are you sure the space on the vmdk is actually free? Also did you run the clear storage bat first successfully?

**Comment by binbon2d** [comment]
yes firs Step Worked
Step 2: 
E:\VMWARE\LDPlayer\ShrinkLDPlayer>"E:\VMWARE\LDPlayer\ShrinkLDPlayer\VdiskManager\vmware-vdiskmanager.exe" -k "E:\VMWARE\LDPlayer\LDPlayer9\vms\leidian1\sdcard.vmdk"
SSLConfigLoad: Failed to load OpenSSL config file.
  Shrink: 100% done.
Shrink completed successfully.

E:\VMWARE\LDPlayer\ShrinkLDPlayer>pause
Press any key to continue . . .

Result: (no change)
![image](https://github.com/user-attachments/assets/b69c8f7d-e025-4232-a5be-cd82116339d4)

**Comment by badjtsx** [comment]
what does it say here?
![Screenshot (162)](https://github.com/user-attachments/assets/10385c13-5047-462e-b2a8-ecd90e79e29a)

**Comment by binbon2d** [comment]
1024 is 1Gb ???
i edit : adb shell "su -c 'cd /sdcard && dd if=/dev/zero of=fillfile bs=1048576 count=10240'"
![image](https://github.com/user-attachments/assets/f0448c33-e2fb-4b40-b202-291431003964)
Befor : 31,986,752

**Comment by badjtsx** [comment]
What I am getting at is that you need to be sure that the vmdk file is too big for what is in the emulator. In other words say you downloaded apps/games on your device but when you deleted them, the vmdk did not shrink in size. Are you sure the emulator does not have 30gb worth of files/apps/games?

**Comment by binbon2d** [comment]
My Disk Info:
![image](https://github.com/user-attachments/assets/53396412-26c7-42ef-b3a3-ca5b9743e33d)

**Comment by binbon2d** [comment]
My game app is about 17-20 GB

**Comment by badjtsx** [comment]
so theoretically the vmdk should shrink down to 20gb?

**Comment by badjtsx** [comment]
Have you tried to do the steps manually? When I was testing it out, doing the steps manually cleared more storage than the bat files

**Comment by binbon2d** [comment]
adb shell "su -c 'cd /sdcard && dd if=/dev/zero of=fillfile bs=1048576 count=1024'"
I tried with 1GB and nothing changed.

The error message
SSLConfigLoad: Failed to load OpenSSL config file.
Is this the cause?

**Comment by badjtsx** [comment]
No it works fine even if openssl fails

**Comment by badjtsx** [comment]
if you're not sure how big the files in your emulator actually are you can use ```adb shell``` and then ```su``` and then ```du -sh```

**Comment by binbon2d** [comment]
Yeah I'll try that later, right now I can't even reduce it by 1GB. I'll let you know when I do.

**Comment by badjtsx** [comment]
Alright then, feel free to ask for help if you run through further problems.

**Comment by binbon2d** [comment]
![image](https://github.com/user-attachments/assets/b919b003-c3e3-42c9-846b-98f42e418c00)
All no such file or directory

**Comment by badjtsx** [comment]
Would you mind showing me what apps you have?

**Comment by badjtsx** [comment]
Also did you try this or has it been the batch file this whole time?
https://github.com/badjtsx/ShrinkLDPlayer?tab=readme-ov-file#manual-version

**Comment by binbon2d** [comment]
Only One **Game Diablo Immortal** download APK from https://www.ldplayer.net/downloads/diablo-immortal-apk.html
and update after time my LD is very big around 40-50GB.
But in reality I think it is only around 17-20G

**Comment by binbon2d** [comment]
![image](https://github.com/user-attachments/assets/2f694be3-7e2f-4f98-b6a1-cb4c42f61457)
27GB

**Comment by badjtsx** [comment]
Try shrinking data then, if the game is 27GB then it's not unlikely the sdcard is 30GB

**Comment by binbon2d** [comment]
At an ID : 0 then there is only
![image](https://github.com/user-attachments/assets/4c65b607-4161-41e4-b87a-5615e1605d35)
I don't understand what kind of Storage mechanism LD has, it's very annoying.

**Comment by badjtsx** [comment]
Just to make sure the batch files are working, can you clone an instance, delete the game completely from that instance, restart it, and then try shrinking it.

**Comment by binbon2d** [comment]
> Just to make sure the batch files are working, can you clone an instance, delete the game completely from that instance, restart it, and then try shrinking it.

OK I will try to agree with you, your work is perfect

**Comment by badjtsx** [comment]
did it work?

**Comment by binbon2d** [comment]
adb shell "su -c 'cd /sdcard && dd if=/dev/zero of=fillfile bs=1048576 count=10240'"
Befor:
![image](https://github.com/user-attachments/assets/6a14c716-01bf-47ca-a40d-a5a12cd081b7)
After:
![image](https://github.com/user-attachments/assets/d83b8f11-df05-48bb-b30d-9b6f6d00f4a7)

IT WORK : I didn't expect this game to be so big.

**Comment by binbon2d** [comment]
Is it require the system disk like this?
![image](https://github.com/user-attachments/assets/b9fee5fe-99aa-4293-9645-a2943b238b8e)
Suddenly, another 2Gb was created

**Comment by badjtsx** [comment]
Not necessary, you can turn it off
